src/proteomes/components/entry/ComponentsButtons.tsx

```diff
@@ -0,0 +1,112 @@
+import { FC, useState, Suspense, useCallback } from 'react';
+import { Button, DownloadIcon } from 'franklin-sites';
+import { Link } from 'react-router-dom';
+
+import SlidingPanel, {
+  Position,
+} from '../../../shared/components/layouts/SlidingPanel';
+
+import useLocalStorage from '../../../shared/hooks/useLocalStorage';
+
+import lazy from '../../../shared/utils/lazy';
+
+import localStorageKeys from '../../../app/config/localStorageKeys';
+import { Column, nsToDefaultColumns } from '../../../shared/config/columns';
+import { LocationToPath, Location } from '../../../app/config/urls';
+
+import { Namespace } from '../../../shared/types/namespaces';
+import { ProteomesAPIModel } from '../../adapters/proteomesConverter';
+
+const DownloadComponent = lazy(
+  () =>
+    import(
+      /* webpackChunkName: "download" */ '../../../shared/components/download/Download'
+    )
+);
+
+const ComponentsButtons: FC<
+  Pick<ProteomesAPIModel, 'id'> & {
+    nComponents: number;
+    selectedEntries: string[];
+  }
+> = ({ id, nComponents, selectedEntries }) => {
+  const [displayDownloadPanel, setDisplayDownloadPanel] = useState(false);
+  const query = '';
+  const defaultColumns = nsToDefaultColumns[Namespace.uniprotkb];
+  const [tableColumns] = useLocalStorage<Column[]>(
+    localStorageKeys.tableColumns(Namespace.uniprotkb),
+    defaultColumns || []
+  );
+
+  const getViewButtonQuery = useCallback(
+    () =>
+      `query=(proteome:${id})${
+        selectedEntries.length !== 0 && selectedEntries.length !== nComponents
+          ? ` AND (${selectedEntries
+              .map((component) => `proteomecomponent:"${component}"`)
+              .join(' OR ')})`
+          : ''
+      }`,
+    [nComponents, id, selectedEntries]
+  );
+
+  const getViewButtonText = useCallback(() => {
+    const nSelected = selectedEntries.length;
+    if (nComponents === 1) {
+      return 'View UniProtKB entry for component';
+    }
+    return `View UniProtKB ${nSelected !== 1 ? 'entries' : 'entry'} for ${
```

Comment:
This is a bit wordy - and even if we don't accomodate for individual component selection we need to reproduce the functionality in the "Map to on the left". Do we have the count breakdown for UniProtKB and TrEMBL in the response? If so the options should be (and consistently accross namespaces when we implement them)

View Proteins (dropdown)
Options:
UniProtKB (count)
Reviewed (Swiss-Prot) (count)
Unreviewed (TrEMBL) (count)

Reply:
>Do we have the count breakdown for UniProtKB and TrEMBL in the response?

Not present in the proteome entry data. If the objective is to create a consistent map to/view proteins experience across the site i would suggest creating a separate component for this that takes a uniprotkb query, makes the requests and populates the dropdown with the counts along with the links.

Reply:
I think this is already provided in the payload for some namespaces so might be an oversight. I'll make a Jira as this is important functionality. For now maybe just shorten to "View proteins" as a button?

Reply:
https://www.ebi.ac.uk/panda/jira/browse/TRM-25668